components/dashboard/src/admin/UserSearch.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { AdminGetListResult, User } from "@gitpod/gitpod-protocol";
+import moment from "moment";
+import { useContext, useEffect, useState } from "react";
+import { useLocation } from "react-router";
+import { Link, Redirect } from "react-router-dom";
+import { PageWithSubMenu } from "../components/PageWithSubMenu";
+import { getGitpodService } from "../service/service";
+import { UserContext } from "../user-context";
+import { adminMenu } from "./admin-menu";
+import UserDetail from "./UserDetail";
+
+export default function UserSearch() {
+    const location = useLocation();
+    const { user } = useContext(UserContext);
+    const [searchResult, setSearchResult] = useState<AdminGetListResult<User>>({ rows: [], total: 0 });
+    const [searchTerm, setSearchTerm] = useState('');
+    const [searching, setSearching] = useState(false);
+    const [currentUser, setCurrentUserState] = useState<User|undefined>(undefined);
+
+    useEffect(() => {
+        const userId = location.pathname.split('/')[3];
+        if (userId) {
+            let user = searchResult.rows.find(u => u.id === userId);
+            if (user) {
+                setCurrentUserState(user);
+            } else {
+                getGitpodService().server.adminGetUser(userId).then(
+                    user => setCurrentUserState(user)
+                ).catch(e => console.error(e));
+            }
+        } else {
+            setCurrentUserState(undefined);
+        }
+    }, [location]);
+
+    if (!user || !user?.rolesOrPermissions?.includes('admin')) {
+        return <Redirect to="/"/>
+    }
+
+    if (currentUser) {
+        return <UserDetail user={currentUser}/>;
+    }
+
+    const search = async () => {
+        setSearching(true);
+        try {
+            const result = await getGitpodService().server.adminGetUsers({
+                searchTerm,
+                limit: 50,
+                orderBy: 'creationDate',
+                offset: 0,
+                orderDir: "desc"
+            });
+            setSearchResult(result);
+        } finally {
+            setSearching(false);
+        }
+    }
+    return <PageWithSubMenu subMenu={adminMenu} title="Users" subtitle="Search an manage all users.">
+        <div className="pt-8 flex">
+            <div className="flex justify-between w-full">
+                <div className="flex">
+                    <div className="py-4">
+                        <svg className={searching ? 'animate-spin' : ''} width="16" height="16" fill="none" xmlns="http://www.w3.org/2000/svg">
```

Comment:
**suggestion**: A spinner before the button label to indicate a search in underway could be better! See [Spin docs](https://tailwindcss.com/docs/animation#spin). Otherwise, let's drop the `animate-spin` here or leave it for fun. Your call! 📞

Reply:
We need indication that search is ongoing.